session/session.go

```diff
@@ -1066,7 +1066,7 @@ func CreateSession(store kv.Storage) (Session, error) {
 	privilege.BindPrivilegeManager(s, pm)
 
 	// Add statsUpdateHandle.
-	if do.StatsHandle() != nil {
+	if do.StatsHandle() != nil && do.StatsUpdating() {
```

Comment:
maybe it's better to add a comment to specify that the data collected by `statsCollector` is consumed by the background stats worker which periodically update stats using the collected data.